### PR TITLE
DAT-3463 handle context-links on Mercury v2.0

### PR DESCRIPTION
**Pull request by dianafa**
https://wikia-inc.atlassian.net/browse/DAT-3463

### CODE REVIEW

File: includes/wikia/parser/templatetypes/TemplateTypesParser.class.php [hunk 1]
```diff
@@ -61,19 +61,20 @@ public static function onGetTemplateDom( $title, $args, $frame, &$outputText ) {
...
 		wfProfileIn( __METHOD__ );
 
-		$title = Title::newFromText( $templateTitle, NS_TEMPLATE );
-
-		if ( self::templateShouldBeProcessed( $templateWikitext ) && self::isValidTitle( $title ) ) {
-			$type = self::getTemplateType( $title );
-			if ( $type == AutomaticTemplateTypes::TEMPLATE_CONTEXT_LINK ) {
-				$templateWikitext = self::handleContextLinksTemplate( $templateWikitext );
+		if ( self::templateShouldBeProcessed( $templateWikitext ) ) {
+				$title = Title::newFromText( $templateTitle, NS_TEMPLATE );
```
Comment (sebastianmarzjan) [inline includes/wikia/parser/templatetypes/TemplateTypesParser.class.php]: This indentation looks odd

File: includes/wikia/parser/templatetypes/handlers/ContextLinkTemplate.class.php [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+class ContextLinkTemplate {
+
+	/**
+	 * @desc check if template should be processed
+	 *
+	 * @param $templateWikitext
+	 *
+	 * @return bool
+	 */
+	public static function templateShouldBeProcessed( $templateWikitext ) {
```
Comment (sebastianmarzjan) [inline includes/wikia/parser/templatetypes/handlers/ContextLinkTemplate.class.php]: I'd suggest reversing the work order (shouldTemplateBeProcessed)

File: includes/wikia/parser/templatetypes/handlers/ContextLinkTemplate.class.php [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+	/**
+	 * @desc sanitize context-link template content
+	 *
+	 * @param $wikitext
+	 *
+	 * @return string
+	 */
+	public static function handle( $wikitext ) {
+		if ( !self::containNestedTemplates( $wikitext ) ) {
```
Comment (sebastianmarzjan) [inline includes/wikia/parser/templatetypes/handlers/ContextLinkTemplate.class.php]: I'm not sure how to read this; What will happen if the context link template DOES contain nested templates?

File: includes/wikia/parser/templatetypes/handlers/ContextLinkTemplate.class.php [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+	/**
+	 * @desc If curly brackets found, means that template contain some not parsed
+	 * wikitext (e.g. was wrongly classified) and should not be changed!
+	 *
+	 * @param $wikitext
+	 * @return bool
+	 */
+	private static function containNestedTemplates( $wikitext ) {
+		return preg_match('/{{.+}}/', $wikitext);
```
Comment (sebastianmarzjan) [inline includes/wikia/parser/templatetypes/handlers/ContextLinkTemplate.class.php]: Are you sure this is non-greedy and will skip arguments?

**Comment by sebastianmarzjan** [comment]
lg

File: includes/wikia/parser/templatetypes/handlers/ContextLinkTemplate.class.php [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+	/**
+	 * @desc sanitize context-link template content
+	 *
+	 * @param $wikitext
+	 *
+	 * @return string
+	 */
+	public static function handle( $wikitext ) {
+			$wikitext = self::sanitizeContextLinkWikitext( $wikitext );
+			$wikitext = self::wrapContextLink( $wikitext );
```
Comment (sebastianmarzjan) [inline includes/wikia/parser/templatetypes/handlers/ContextLinkTemplate.class.php]: Awkward tabs detected :)

File: includes/wikia/parser/templatetypes/handlers/ContextLinkTemplate.class.php [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+	 *
+	 * @param $templateWikitext
+	 *
+	 * @return bool
+	 */
+	public static function shouldTemplateBeProcessed( $templateWikitext ) {
+		global $wgEnableTemplateTypesParsing, $wgArticleAsJson;
+
+		return $wgEnableTemplateTypesParsing
+		&& $wgArticleAsJson
```
Comment (sebastianmarzjan) [inline includes/wikia/parser/templatetypes/handlers/ContextLinkTemplate.class.php]: Indentation seems strange here too

**Comment by sebastianmarzjan** [comment]
Please take a look at the comments and merge latest parent branch to this one to mitigate conflicts

**Comment by sebastianmarzjan** [comment]
LGTM

**Comment by dianafa** [comment]
created a new PR to dev branch: https://github.com/Wikia/app/pull/9133

**Pull request closed without merging**